packages/grafana-runtime/src/services/SidecarService_EXPERIMENTAL.ts

```diff
@@ -74,39 +74,48 @@ export class SidecarService_EXPERIMENTAL {
    * @experimental
    */
   get activePluginId() {
-    return this._activePluginId.getValue();
+    return getPluginIdFromUrl(this.memoryLocationService.getLocation().pathname);
+  }
+
+  getLocationService() {
+    return this.memoryLocationService;
   }
 
   /**
    * Opens an app in a sidecar. You can also pass some context object that will be then available to the app.
+   * @deprecated
    * @experimental
    */
   openApp(pluginId: string, context?: unknown) {
     if (!this.assertFeatureEnabled()) {
       return;
     }
-    if (this.localStorageKey) {
-      localStorage.setItem(this.localStorageKey, pluginId);
-    }
-
-    this._activePluginId.next(pluginId);
     this._initialContext.next(context);
+    this.memoryLocationService.push({ pathname: `/a/${pluginId}` });
   }
 
   /**
+   * Opens an app in a sidecar. You can also relative path inside the app to open.
    * @experimental
    */
-  closeApp(pluginId: string) {
+  openAppV2(pluginId: string, path?: string) {
```

Comment:
The idea is to keep both for now until this stabilises and then go and replace everything but don't want to go and do all the app PRs for this right now and don't want to break what ever works so far.